File: Problems/Algorithms/435. Non-Overlapping Intervals/NonOverlappingIntervals.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        int n = intervals.length;
        if (n <= 1) return 0;
        
        Arrays.sort(intervals, (a, b) -> a[1] - b[1]);
        
        int ans = 0;
        int i = 0;
        int j = 1;
        while (j < n) {
            if (intervals[j][0] < intervals[i][1]) {
                ans++;
            } else {
                i = j;
            }
            j++;
        }
        
        return ans;
    }
}

File: Problems/Algorithms/435. Non-Overlapping Intervals/non-overlapping-intervals.cpp
#include <vector>

using namespace std;

bool comparator(vector<int>& a, vector<int>& b) {
    return a[1] < b[1];
}

class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), comparator);

        int ans = 0;
        int last = INT_MIN;

        for (int i = 0; i < intervals.size(); i++) {
            int x = intervals[i][0], y = intervals[i][1];

            if (x >= last) {
                last = y;
            } else {
                ans++;
            }
        }

        return ans;
    }
};
